awss3/s3_bucket.go

```diff
@@ -185,22 +187,25 @@ func (s *S3Bucket) checkDeletePublicAccessBlock(bucketDetails BucketDetails, buc
 	publicAccessPolicy := bucketPolicyStatement{
 		Effect:    "Allow",
 		Principal: "*",
-		Action:    []string{"s3.GetObject"},
+		Action:    []string{"s3:GetObject"},
 	}
-
 	if slices.ContainsFunc(policy.Statement, func(statement bucketPolicyStatement) bool {
 		return statement.Effect == publicAccessPolicy.Effect &&
 			statement.Principal == publicAccessPolicy.Principal &&
 			slices.Equal(statement.Action, publicAccessPolicy.Action)
 	}) {
-		_, err := s.s3svc.DeletePublicAccessBlock(&s3.DeletePublicAccessBlockInput{
+		deletePublicAccessBlockInput := &s3.DeletePublicAccessBlockInput{
 			Bucket: aws.String(bucketName),
-		})
+		}
+		s.logger.Debug("delete-public-access-block", lager.Data{"input": deletePublicAccessBlockInput})
+		_, err := s.s3svc.DeletePublicAccessBlock(deletePublicAccessBlockInput)
 		if err != nil {
-			s.logger.Error("aws-s3-error", err)
+			s.logger.Error("failed to delete public access block", err)
 			return err
 		}
 	}
+	// sleep to ensure that public access block is deleted before proceeding to put bucket polcy
```

Comment:
I don't want to hold this up right now, but can we add a TODO to change this from a sleep to checking with AWS's API to see if the block has been removed?